### ISSUE TITLE
NVDA doesn't announce typed characters in putty

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#1348]
**Reported by aleksey_s on 2011-02-02 17:17**
It neither reads typed characters nor announces a deleted character when pressing backspace in putty's terminal window.

**Comment by leonardder** [comment]
Comfirmed. Since this is quite an old issue, giving this P4. @JCSTeh: Feel free to set another priority, since you are the initial author of this.

**Comment by leonardder** [comment]
I noticed that this is also a problem in power shell. Given Power Shell is getting more and more a replacement for cmd, I wonder whether this priority should be raised. @michaeldcurran/@feerrenrut: thoughts about this?

**Comment by feerrenrut** [comment]
Yep, I'll apply that change now.

**Comment by Adriani90** [comment]
@aleksey_s, @Mohammed, could you please test if this is still an issue? Thanks!

**Comment by leonardder** [comment]
Yes, this is still an issue.

**Comment by JulienCochuyt** [comment]
MinTTY, a close relative console, also suffers from this issue: Typed characters are not echoed.
The `mintty` app module derives from the `putty` app module.
MinTTY is the default console of the Git Bash environment as shipped with Git for Windows.

**Comment by Adriani90** [comment]
Is this solved also for older Versions of Windows where UIA is not used?

**Comment by JulienCochuyt** [comment]
> Is this solved also for older Versions of Windows where UIA is not used?

I don't know, but IMHO, this is not fully solved either for newer versions of Windows.
Demo: Open a new Git Bash Mintty terminal, run any command that produces output such as `ls`, wait for the output announcement to end or hit control to mute it, and then hit the backspace key a few times quickly.
Result: NVDA speaks text placed before the command prompt, depending on how many times backspace was pressed.
Expected: NVDA should not announce anything, as no character was deleted.

**Comment by Adriani90** [comment]
Ok Julien thank you for this report. I am reopening this issue and the discussion.
cc: @codeofdusk

**Comment by codeofdusk** [comment]
> Is this solved also for older Versions of Windows where UIA is not used?

Yes, this is solved from Windows 10 1607 onwards (UIA used on 1803 and later).

**Comment by codeofdusk** [comment]
> I don't know, but IMHO, this is not fully solved either for newer versions of Windows.
> Demo: Open a new Git Bash Mintty terminal, run any command that produces output such as `ls`, wait for the output announcement to end or hit control to mute it, and then hit the backspace key a few times quickly.
> Result: NVDA speaks text placed before the command prompt, depending on how many times backspace was pressed.
> Expected: NVDA should not announce anything, as no character was deleted.

I think I know what's up (related to #9933). I've put a potential fix on the t1348 branch of my repo (@feerrenrut could you please create and link a try build here?)
